knowledge-base/implementation-documentation/api-improvements.md

```diff
@@ -0,0 +1,209 @@
+# Potential API improvements
+
+This document contains some ideas for API changes, that would be a breaking change and would result in a major version upgrade.
+
+
+## Filtering root elements based on 1-1 relationships
+
+### Current API
+```ts
+BusinessPartner.requestBuilder().getAll()
+  .filter(
+    BusinessPartner.TO_BUSINESS_PARTNER_ADDRESS.filter(
+      BusinessPartnerAddress.CITY.equals('Berlin')
+    )
+  )
+```
+### Issues
+
+The inner filter is quite confusing, because `filter` implies that we are filtering the Address here, while it is actually filtering the business partner.
+
+### Alternatives
+
+`filterBy`
+
+```ts
+BusinessPartner.requestBuilder().getAll()
+  .filter(
+    BusinessPartner.TO_BUSINESS_PARTNER_ADDRESS.filterBy(
+      BusinessPartnerAddress.CITY.equals('Berlin')
+    )
+  )
+```
+
```

Comment:
```suggestion
Concerns:
* The difference in language `filter` versus `filterBy` is too subtle for me. I would just overlook it. Also since it is not directly on the root entity like: `BusinessPartner.filterBy`
```

Reply:
Can you formulate that without using "me"? I disagree, that the difference in language is too subtle, but I agree, that it is not 100 % clear what is meant by that, because the context does not disclose that.